impl/client/src/main/javascript/web/prompting/components/ParameterPanelComponent.js

```diff
@@ -26,6 +26,23 @@ define("common-ui/prompting/components/ParameterPanelComponent", ['./PanelCompon
     removeErrorClass: function(){
       var $htmlObject = this.placeholder();
       $htmlObject.removeClass('error');
+    },
+
+    addErrorClass: function(){
+      var $htmlObject = this.placeholder();
+      $htmlObject.addClass('error');
+    },
+
+    addErrorLabel: function(errComponent, errMessage){
+      var $htmlObject = this.placeholder();
+      if(!errComponent || !errMessage)
+        return;
+
+      var errorLabelHtml = '<div id="' + errComponent.htmlObject + '" class="parameter-label">';
+      errorLabelHtml += errMessage + '</div>';
```

Comment:
I don't think you should be injecting the error message like this. Is it sanitized? How is it done in the full-render mode?

Reply:
In full render mode is done like is described in the following example js https://github.com/pentaho/pentaho-platform-plugin-common-ui/blob/master/impl/client/src/main/javascript/web/prompting/components/CompositeComponent.js, you don't have a function receiving directly an error message sting. But as we discussed previously, this would be a very "special" case of creating and injecting an label on a panel component in opposite to all CDF logic where all components were static, that why I opted by this solution. In your opinion what would be the ideal solution?

Reply:
You should be creating an empty div for the provided errComponent. It is the component that, when updated, should fill-in the div with the error message. Don't know how the specific component receives the error message.
You need to update the component afterwards.